## Dataset de Usuarios

### Limpieza inicial

In [1]:
import pandas as pd

df_users = pd.read_csv('Dataset/datos_usuarios.csv')

/home/david/.local/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [2]:
df_users

usuario_id   nombre  num_opiniones        usuario_desde  \
0        l6BmjZMeQD3rDxWUbiAiow   Rashmi             95  2013-10-08 23:11:33   
1        4XChL029mKr5hydo79Ljxg    Jenna             33  2013-02-21 22:29:06   
2        bc8C_eETBWL0olvFSJJd0w    David             16  2013-10-04 00:16:10   
3        dD0gZpBctWGdWo9WlGuhlA   Angela             17  2014-05-22 15:57:30   
4        MM4RJAeH6yuaN8oZDSt0RA    Nancy            361  2013-10-23 07:02:50   
...                         ...      ...            ...                  ...   
1637133  oC4BjbB4I-wx5KBAO1k2Gg   Jiaxin              2  2016-02-24 19:28:47   
1637134  LKd_-BT2GMWp8_1ho18kOg      Wes              8  2017-02-25 23:34:05   
1637135  ib_elCBhGOamS0AALkh4Og      Lee              4  2009-08-19 22:08:16   
1637136  WV27LoSFDG_YmiDKTLKKLA  Melissa              1  2018-10-15 17:07:43   
1637137  agVqH9xDhNw8n7ve66H1Uw     Erin              1  2016-03-03 21:56:06   

         util  gracioso  guay              experto  \
0          84        17    25       2015,2016,2017   
1          48        22    16                  NaN   
2          28         8    10                  NaN   
3          30         4    14                  NaN   
4        1114       279   665  2015,2016,2017,2018   
...       ...       ...   ...                  ...   
1637133     0         0     0                  NaN   
1637134     0         0     0                  NaN   
1637135     8         0     0                  NaN   
1637136     1         0     0                  NaN   
1637137     0         0     0                  NaN   

                                                    amigos  seguidores  ...  \
0        c78V-rj8NQcQjOI8KP3UEA, alRMgPcngYSCJ5naFRBz5g...           5  ...   
1        kEBTgDvFX754S68FllfCaA, aB2DynOxNOJK9st2ZeGTPg...           4  ...   
2        4N-HU_T32hLENLntsNKNBg, pSY2vwWLgWfGVAAiKQzMng...           0  ...   
3        RZ6wS38wnlXyj-OOdTzBxA, l5jxZh1KsgI8rMunm-GN6A...           5  ...   
4        mbwrZ-RS76V1HoJ0bF_Geg, g64lOV39xSLRZO0aQQ6DeQ...          39  ...   
...                                                    ...         ...  ...   
1637133                                               None           0  ...   
1637134                                               None           0  ...   
1637135                                               None           0  ...   
1637136                                               None           0  ...   
1637137                                               None           0  ...   

         like_extras  like_perfil  like_formato  like_lista  like_comentario  \
0                  0            0             0           0                1   
1                  0            0             0           0                0   
2                  0            0             0           0                1   
3                  0            0             0           0                0   
4                  1            0             0           1               16   
...              ...          ...           ...         ...              ...   
1637133            0            0             0           0                0   
1637134            0            0             0           0                0   
1637135            0            0             0           0                0   
1637136            0            0             0           0                0   
1637137            0            0             0           0                0   

         like_simple  like_guay  like_divertido  like_textos  like_fotos  
0                  1          1               1            2           0  
1                  0          1               1            0           0  
2                  0          0               0            0           0  
3                  2          0               0            1           0  
4                 57         80              80           25           5  
...              ...        ...             ...    

### Búsqueda de valores nulos

In [3]:
print("¿Hay valores nulos?\n", pd.isnull(df_users).any(), "\n")
print("¿Cuantos?\n", pd.isnull(df_users).sum(), "\n")

¿Hay valores nulos?
 usuario_id          False
nombre               True
num_opiniones       False
usuario_desde       False
util                False
gracioso            False
guay                False
experto              True
amigos              False
seguidores          False
media_puntuacion    False
like_moda           False
like_extras         False
like_perfil         False
like_formato        False
like_lista          False
like_comentario     False
like_simple         False
like_guay           False
like_divertido      False
like_textos         False
like_fotos          False
dtype: bool 

¿Cuantos?
 usuario_id                0
nombre                    3
num_opiniones             0
usuario_desde             0
util                      0
gracioso                  0
guay                      0
experto             1565761
amigos                    0
seguidores                0
media_puntuacion          0
like_moda                 0
like_extras               0
like_perfil       

Se han encontrado algunos valores nulos en el atributo nombre y muchísimos en el atributo "experto". En este último atributo se tienen los años en el que el usuario ha sido experto, por lo que muchos usuarios no han sido nunca expertos.

Decidimos completar los nombres nulos de nombre con "SinNombre" y los valores nulos de experto con "Ninguno". No eliminamos las instancias de usuarios sin nombre porque podemos seguir identificándolo con su ID. Tambien cambiar None de un usuario sin amigos por "SinAmigos", para tener controlado e identificado ese caso.

In [4]:
# Rellenamos valores nulos y NaN con un valor específico y controlado
df_users['experto'] = df_users['experto'].fillna("Ninguno")
df_users['nombre'] = df_users['nombre'].fillna("SinNombre")
df_users['amigos'] = df_users['amigos'].replace("None", "SinAmigos")
df_users

usuario_id   nombre  num_opiniones        usuario_desde  \
0        l6BmjZMeQD3rDxWUbiAiow   Rashmi             95  2013-10-08 23:11:33   
1        4XChL029mKr5hydo79Ljxg    Jenna             33  2013-02-21 22:29:06   
2        bc8C_eETBWL0olvFSJJd0w    David             16  2013-10-04 00:16:10   
3        dD0gZpBctWGdWo9WlGuhlA   Angela             17  2014-05-22 15:57:30   
4        MM4RJAeH6yuaN8oZDSt0RA    Nancy            361  2013-10-23 07:02:50   
...                         ...      ...            ...                  ...   
1637133  oC4BjbB4I-wx5KBAO1k2Gg   Jiaxin              2  2016-02-24 19:28:47   
1637134  LKd_-BT2GMWp8_1ho18kOg      Wes              8  2017-02-25 23:34:05   
1637135  ib_elCBhGOamS0AALkh4Og      Lee              4  2009-08-19 22:08:16   
1637136  WV27LoSFDG_YmiDKTLKKLA  Melissa              1  2018-10-15 17:07:43   
1637137  agVqH9xDhNw8n7ve66H1Uw     Erin              1  2016-03-03 21:56:06   

         util  gracioso  guay              experto  \
0          84        17    25       2015,2016,2017   
1          48        22    16              Ninguno   
2          28         8    10              Ninguno   
3          30         4    14              Ninguno   
4        1114       279   665  2015,2016,2017,2018   
...       ...       ...   ...                  ...   
1637133     0         0     0              Ninguno   
1637134     0         0     0              Ninguno   
1637135     8         0     0              Ninguno   
1637136     1         0     0              Ninguno   
1637137     0         0     0              Ninguno   

                                                    amigos  seguidores  ...  \
0        c78V-rj8NQcQjOI8KP3UEA, alRMgPcngYSCJ5naFRBz5g...           5  ...   
1        kEBTgDvFX754S68FllfCaA, aB2DynOxNOJK9st2ZeGTPg...           4  ...   
2        4N-HU_T32hLENLntsNKNBg, pSY2vwWLgWfGVAAiKQzMng...           0  ...   
3        RZ6wS38wnlXyj-OOdTzBxA, l5jxZh1KsgI8rMunm-GN6A...           5  ...   
4        mbwrZ-RS76V1HoJ0bF_Geg, g64lOV39xSLRZO0aQQ6DeQ...          39  ...   
...                                                    ...         ...  ...   
1637133                                          SinAmigos           0  ...   
1637134                                          SinAmigos           0  ...   
1637135                                          SinAmigos           0  ...   
1637136                                          SinAmigos           0  ...   
1637137                                          SinAmigos           0  ...   

         like_extras  like_perfil  like_formato  like_lista  like_comentario  \
0                  0            0             0           0                1   
1                  0            0             0           0                0   
2                  0            0             0           0                1   
3                  0            0             0           0                0   
4                  1            0             0           1               16   
...              ...          ...           ...         ...              ...   
1637133            0            0             0           0                0   
1637134            0            0             0           0                0   
1637135            0            0             0           0                0   
1637136            0            0             0           0                0   
1637137            0            0             0           0                0   

         like_simple  like_guay  like_divertido  like_textos  like_fotos  
0                  1          1               1            2           0  
1                  0          1               1            0           0  
2                  0          0               0            0           0  
3                  2          0               0            1           0  
4                 57         80              80           25           5  
...              ...        ...             ...    

In [5]:
print("¿Hay valores nulos?\n", pd.isnull(df_users).any(), "\n")
print("¿Cuantos?\n", pd.isnull(df_users).sum(), "\n")

¿Hay valores nulos?
 usuario_id          False
nombre              False
num_opiniones       False
usuario_desde       False
util                False
gracioso            False
guay                False
experto             False
amigos              False
seguidores          False
media_puntuacion    False
like_moda           False
like_extras         False
like_perfil         False
like_formato        False
like_lista          False
like_comentario     False
like_simple         False
like_guay           False
like_divertido      False
like_textos         False
like_fotos          False
dtype: bool 

¿Cuantos?
 usuario_id          0
nombre              0
num_opiniones       0
usuario_desde       0
util                0
gracioso            0
guay                0
experto             0
amigos              0
seguidores          0
media_puntuacion    0
like_moda           0
like_extras         0
like_perfil         0
like_formato        0
like_lista          0
like_comentario     0
like_sim

### Búsqueda de inconsistencias en los datos

Algo que nos ha parecido interesante comprobar es si los amigos de cada usuario están en la base datos, hemos decidido hacer esta comprobación porque nos parecía algo que debía ser consistente.

In [6]:
users_id = df_users["usuario_id"]

# Recorremos las listas de amigos de cada usuario para comprobar si existen
for amigos in df_users["amigos"][0:1]:
    listaAmigos = amigos.replace(" ", "")
    listaAmigos = listaAmigos.split(",")
    numAmigos = len(listaAmigos)
    
    exists = df_users["usuario_id"].isin(listaAmigos)
    
    print("Numero de amigos: " + str(numAmigos))
    print(exists.value_counts())

Numero de amigos: 99
False    1637125
True          13
Name: usuario_id, dtype: int64


Comprobamos solo para el primer usuario ya que imprimir todos es demasiado, pero ocurre lo mismo con todos. En el caso impreso, el usuario tiene 99 amigos de los cuales solo 13 están en la base de datos. Ya que inicialmente esto nos parece muy extraño, decidimos imprimir cuales están y cuales no para comprobarlo manualmente al menos para un par de ejemplos:

In [7]:
users_id = df_users["usuario_id"]

# Recorremos las listas de amigo de cada usuario imprimiendo para cada user si existe o no
for amigos in df_users["amigos"][0:1]:
    listaAmigos = amigos.replace(" ", "")
    listaAmigos = listaAmigos.split(",")
    numAmigos = len(listaAmigos)
    for amigo in listaAmigos:
        lista = []
        lista.append(amigo)
        exists = df_users["usuario_id"].isin(lista).to_numpy()

        if True in exists:
            print(amigo + ": True")
        else:
            print(amigo + ": False")

c78V-rj8NQcQjOI8KP3UEA: False
alRMgPcngYSCJ5naFRBz5g: False
ajcnq75Z5xxkvUSmmJ1bCg: False
BSMAmp2-wMzCkhTfq9ToNg: False
jka10dk9ygX76hJG0gfPZQ: False
dut0e4xvme7QSlesOycHQA: False
l4l5lBnK356zBua7B-UJ6Q: False
0HicMOOs-M_gl2eO-zES4Q: True
_uI57wL2fLyftrcSFpfSGQ: False
T4_Qd0YWbC3co6WSMw4vxg: False
iBRoLWPtWmsI1kdbE9ORSA: False
xjrUcid6Ymq0DoTJELkYyw: False
GqadWVzJ6At-vgLzK_SKgA: False
DvB13VJBmSnbFXBVBsKmDA: False
vRP9nQkYTeNioDjtxZlVhg: True
gT0A1iN3eeQ8EMAjJhwQtw: False
6yCWjFPtp_AD4x93WAwmnw: False
1dKzpNnib-JlViKv8_Gt5g: False
3Bv4_JxHXq-gVLOxYMQX0Q: False
ikQyfu1iViYh8T0us7wiFQ: False
f1GGltNaB7K5DR1jf3dOmg: False
tgeFUChlh7v8bZFVl2-hjQ: False
-9-9oyXlqsMG2he5xIWdLQ: False
Adj9fBPVJad8vSs-mIP7gw: False
Ce49RY8CKXVsTifxRYFTsw: True
M1_7TLi8CbdA89nFLlH4iw: True
wFsNv-hqbW_F5-IRqfBN6g: False
0Q1L7zXHocaUZ2gsG2XJeg: False
cBFgmOCBdhYa0xoFEAzp_g: True
VrD_AgiFvzqtlR15vir3SQ: True
cpE-7HK514Sr5vpSen9CEQ: False
F1UYelhPFB-zIKlt0ygIZg: False
CQAL1hvsLMCzuJf9AglsXw: False
1KnY1wr15WfEWIRL

In [8]:
# Comprobamos este usuario: f1GGltNaB7K5DR1jf3dOmg: False
print(df_users.loc[df_users["usuario_id"] == "f1GGltNaB7K5DR1jf3dOmg"])

#Comprobamos este usuario: VrD_AgiFvzqtlR15vir3SQ: True
df_users.loc[df_users["usuario_id"] == "VrD_AgiFvzqtlR15vir3SQ"]

Empty DataFrame
Columns: [usuario_id, nombre, num_opiniones, usuario_desde, util, gracioso, guay, experto, amigos, seguidores, media_puntuacion, like_moda, like_extras, like_perfil, like_formato, like_lista, like_comentario, like_simple, like_guay, like_divertido, like_textos, like_fotos]
Index: []

[0 rows x 22 columns]


usuario_id  nombre  num_opiniones        usuario_desde  \
216258  VrD_AgiFvzqtlR15vir3SQ  Vedran             58  2014-07-12 15:09:57   

        util  gracioso  guay  experto  \
216258    41         9     5  Ninguno   

                                                   amigos  seguidores  ...  \
216258  qiYcRBQVMwEdITGLJPcDRg, PxPN0tS9qg5gfg-HIv48gQ...           1  ...   

        like_extras  like_perfil  like_formato  like_lista  like_comentario  \
216258            0            0             0           0                4   

        like_simple  like_guay  like_divertido  like_textos  like_fotos  
216258            2          0               0            0           0  

[1 rows x 22 columns]

Como podemos ver, algunos usuarios amigos no se encuentran en la base de datos. Esto se debe a que no tenemos la base de datos completa de los usuarios de la plataforma, por lo que decidimos dejarlo tal cual lo tenemos. Una opción podría ser eliminar todos los usuarios de la lista de amigos que no existen en la base de datos, pero no lo consideramos necesario (además de que según la forma en la que lo estábamos haciendo, necesita demasiado tiempo de cómputo)

### Comprobación sintáctica de la fecha

Comprobamos si la fecha es correcta sintácticamente utilizando una expresión regular. No se imprime ninguna fecha incorrecta, por lo que todas están bien.

In [9]:
import re
pattern = re.compile("[0-9][0-9][0-9][0-9][-][0-1][0-9][-][0-3][0-9]\s[0-2][0-9][:][0-5][0-9][:][0-5][0-9]")


def preprocesa(x):
    if(pattern.match(x) == None):
        print("date not matching: " + str(x))
        return True
    return False

df_users['usuario_desde'].apply(lambda x: preprocesa(x))

0          False
1          False
2          False
3          False
4          False
           ...  
1637133    False
1637134    False
1637135    False
1637136    False
1637137    False
Name: usuario_desde, Length: 1637138, dtype: bool

### Transformación de los datos
De todos los elementos que podemos quedarnos de la fecha, decidimos quedarnos con el día, el mes, el añoy la hora ya que puede ser que posteriormente nos sean útiles. El atributo que puede llegar a ser más controversial es el minuto puesto que se puede considerar que es hilar demasiado fino. Los atributos con los que nos quedamos, nos pueden servir para hacer análisis de los registros que se realizan. Hemos tenido dudas a la hroa de quedarnos con la hora, pero nos podría servir, por ejemplo, para determinar a qué hora se realizan más registros e informar a nuestro cliente (podría ser necesario un refuerzo en los servidores a una hora determinada).

In [10]:
# Dividimos la usuario_desde en año, mes , dia, hora y minuto
df_users['anyo_reg'] = pd.DatetimeIndex(df_users['usuario_desde']).year
df_users['mes_reg'] = pd.DatetimeIndex(df_users['usuario_desde']).month
df_users['dia_reg'] = pd.DatetimeIndex(df_users['usuario_desde']).day
df_users['hora_reg'] = pd.DatetimeIndex(df_users['usuario_desde']).hour

# Eliminamos el atributo usuario_desde
df_users = df_users.drop("usuario_desde", axis=1)
df_users

usuario_id   nombre  num_opiniones  util  gracioso  guay  \
0        l6BmjZMeQD3rDxWUbiAiow   Rashmi             95    84        17    25   
1        4XChL029mKr5hydo79Ljxg    Jenna             33    48        22    16   
2        bc8C_eETBWL0olvFSJJd0w    David             16    28         8    10   
3        dD0gZpBctWGdWo9WlGuhlA   Angela             17    30         4    14   
4        MM4RJAeH6yuaN8oZDSt0RA    Nancy            361  1114       279   665   
...                         ...      ...            ...   ...       ...   ...   
1637133  oC4BjbB4I-wx5KBAO1k2Gg   Jiaxin              2     0         0     0   
1637134  LKd_-BT2GMWp8_1ho18kOg      Wes              8     0         0     0   
1637135  ib_elCBhGOamS0AALkh4Og      Lee              4     8         0     0   
1637136  WV27LoSFDG_YmiDKTLKKLA  Melissa              1     1         0     0   
1637137  agVqH9xDhNw8n7ve66H1Uw     Erin              1     0         0     0   

                     experto  \
0             2015,2016,2017   
1                    Ninguno   
2                    Ninguno   
3                    Ninguno   
4        2015,2016,2017,2018   
...                      ...   
1637133              Ninguno   
1637134              Ninguno   
1637135              Ninguno   
1637136              Ninguno   
1637137              Ninguno   

                                                    amigos  seguidores  \
0        c78V-rj8NQcQjOI8KP3UEA, alRMgPcngYSCJ5naFRBz5g...           5   
1        kEBTgDvFX754S68FllfCaA, aB2DynOxNOJK9st2ZeGTPg...           4   
2        4N-HU_T32hLENLntsNKNBg, pSY2vwWLgWfGVAAiKQzMng...           0   
3        RZ6wS38wnlXyj-OOdTzBxA, l5jxZh1KsgI8rMunm-GN6A...           5   
4        mbwrZ-RS76V1HoJ0bF_Geg, g64lOV39xSLRZO0aQQ6DeQ...          39   
...                                                    ...         ...   
1637133                                          SinAmigos           0   
1637134                                          SinAmigos           0   
1637135                                          SinAmigos           0   
1637136                                          SinAmigos           0   
1637137                                          SinAmigos           0   

         media_puntuacion  ...  like_comentario  like_simple  like_guay  \
0                    4.03  ...                1            1          1   
1                    3.63  ...                0            0          1   
2                    3.71  ...                1            0          0   
3                    4.85  ...                0            2          0   
4                    4.08  ...               16           57         80   
...                   ...  ...              ...          ...        ...   
1637133              4.50  ...                0            0          0   
1637134              4.00  ...                0            0          0   
1637135              3.40  ...                0            0          0   
1637136              3.00  ...                0            0          0   
1637137              5.00  ...                0            0          0   

         like_divertido  like_textos  like_fotos  anyo_reg  mes_reg  dia_reg  \
0                     1            2           0      2013       10        8   
1                     1            0           0      2013        2       21   
2                     0            0           0      2013       10        4   
3                     0            1           0      2014        5       22   
4                    80           25           5      2013       10       23   
...                 ...          ...         ...       ...      ...      ...   
1637133               0            0           0      2016        2       24   
1637134               0            0           0      2017        2       25   
1637135               0            0           0      2009        8       19   
1637136               0            0           0      2018       10      

### Número de amigos

Aunque tenemos en un array qué amigos tiene un usuario y si no tiene amigos, vamos a incluir una nueva columna cuyo objetivo sea indicarnos la cantidad concreta de amigos que tiene cada usuario.

In [11]:
def genNumAmigos (row):
    amigos = row['amigos'].split(',')
    numAmigos = len(amigos)
    if numAmigos == 1 and amigos[0] == 'SinAmigos':
        numAmigos = 0
        
    return numAmigos

df_users["num_amigos"] = df_users.apply(lambda row: genNumAmigos(row), axis = 1)

In [12]:
cols = df_users.columns.tolist()
cols = cols[:8] + cols[-1:] + cols[8:-1]

df_users = df_users[cols]

In [13]:
df_users["num_amigos"]

0            99
1          1152
2            15
3           525
4           231
           ... 
1637133       0
1637134       0
1637135       0
1637136       0
1637137       0
Name: num_amigos, Length: 1637138, dtype: int64

### Comprobación de distribución de valores
En primer lugar, vamos a hacer un describe para comprobar que el rango de valores de los atributos tiene sentido. El objetivo es encontrar cosas como valores extraños (negativos o desorbitados) en likes, fechas... etc

In [14]:
# Imprimimos todas las columnas del describe
pd.set_option('display.max_columns', 500)

# Lo ponemos de esta forma para quitar la notación científica
df_users.describe().apply(lambda s: s.apply(lambda x: format(x, 'g')))

num_opiniones         util     gracioso         guay   num_amigos  \
count   1.63714e+06  1.63714e+06  1.63714e+06  1.63714e+06  1.63714e+06   
mean        22.2877      40.5277      17.9178      21.8805      45.7152   
std         76.9224      470.242       330.35      401.968      137.863   
min               0            0            0            0            0   
25%               2            0            0            0            0   
50%               5            3            0            0            2   
75%              15           13            3            3           39   
max           13278       154202       130207       148658        14995   

        seguidores media_puntuacion    like_moda  like_extras  like_perfil  \
count  1.63714e+06      1.63714e+06  1.63714e+06  1.63714e+06  1.63714e+06   
mean       1.46565          3.68146      2.24223     0.332595     0.213513   
std        15.6832          1.15104      80.2182      13.6785      16.2668   
min              0                1            0            0            0   
25%              0                3            0            0            0   
50%              0             3.89            0            0            0   
75%              0              4.6            0            0            0   
max           9538                5        34167        13500        14173   

      like_formato   like_lista like_comentario  like_simple    like_guay  \
count  1.63714e+06  1.63714e+06     1.63714e+06  1.63714e+06  1.63714e+06   
mean      0.190208    0.0835849         1.50394      3.09027      3.10682   
std         12.698      10.8956          63.276      92.1142      88.2326   
min              0            0               0            0            0   
25%              0            0               0            0            0   
50%              0            0               0            0            0   
75%              0            0               0            0            0   
max          13654        12669           57833        52103        32266   

      like_divertido  like_textos   like_fotos     anyo_reg      mes_reg  \
count    1.63714e+06  1.63714e+06  1.63714e+06  1.63714e+06  1.63714e+06   
mean         3.10682      1.16533      1.13843      2013.47       6.4201   
std          88.2326      32.6826      95.4914      2.60379      3.36132   
min                0            0            0         2004            1   
25%                0            0            0         2012            4   
50%                0            0            0         2014            6   
75%                0            0            0         2015            9   
max            32266        15442        82602         2018           12   

           dia_reg     hora_reg  
count  1.63714e+06  1.63714e+06  
mean       15.7372      12.5857  
std        8.80282      8.11387  
min              1            0  
25%              8            4  
50%             16           16  
75%             23           20  
max             31           23

Como se puede observar en los mínimos presentados en la tabla de análisis descriptivo, no aparece ningún número que sea menor que cero y que, por lo tanto, no tenga sentido. Sin embargo, sí podemos ver algo que nos ha llamado bastante la atención; la media del número de likes. Como se puede comprobar, la media de la cantidad de likes (de todos los tipos de likes) que se dan es muy cercana a 0 indicando que la mayor parte de las opiniones presentadas por la gente no reciben like. Ahora, si miramos a los máximos podemos comprobar que son realmente elevados. Esto nos está indicando que existen unas pocas opiniones que reciben una gran cantidad de likes y que, posiblemente, estén relacionadas con personas con muchos seguidores; los influencer. Creemos que durante el análisis será interesante hablar más de estos usuarios ya que pueden suponer un gran beneficio para un presunto cliente.

Algo que también merece la pena destacar es que las columnas "util", "gracioso" y "guay", las que llamaremos "impresiones", aunque deberían tener unas características similares a las de los likes, son bastante más utilizadas por los usuarios y tienen una media un tanto más elevada.

Por último, nos gustaría destacar que con los seguidores ocurre lo mismo que con los likes. La mayor parte de los usuarios no tienen seguidores pero hay unos pocos que tienen una gran cantidad (alcanzando un máximo de 9538).

In [15]:
df_users.to_csv("datos_users_limpio.csv", index=False)